Domain/ImageUpdater.cs

```diff
@@ -4,59 +4,65 @@
         private readonly Regex _tagFilter;
         private readonly IK8sApi _k8sApi;
 
-        public ImageUpdater(ILogger<ImageUpdater> logger, UpdaterDbContext context, IOptions<AppSettings> settings, IK8sApi k8sApi)
+        public ImageUpdater(ILogger<ImageUpdater> logger, IOptions<AppSettings> settings, IK8sApi k8sApi)
         {
             _logger = logger;
-            _context = context;
             _tagFilter = new Regex(settings.Value.UpdateTagsMatching);
             _k8sApi = k8sApi;
         }
 
-        public IEnumerable<ImageEvent> UpdateEventHandler(string newImageUri)
+        public IEnumerable<ImageEvent> UpdateEventHandler(string imageToUpdateUri)
         {
-            var parsedUri = ImageUriParser.ParseUri(newImageUri);
+            var parsedUri = ImageUriParser.ParseUri(imageToUpdateUri);
 
-            return _k8sApi.GetImages()
-                .Where(currentClusterImage => CheckIfImageIsApplicapleForDeployment(currentClusterImage, parsedUri))
-                .Select(image => SetNewImage(parsedUri, image))
-                .ToList();
-        }
-
-        private bool CheckIfImageIsApplicapleForDeployment(ImageInCluster currentClusterImage, (string uri, string tag) parsedUri)
-        {
-            var isApplicaple = _tagFilter.IsMatch(ImageUriParser.ParseUri(currentClusterImage.Image).tag) && ImageUriParser.ParseUri(currentClusterImage.Image).uri == parsedUri.uri;
+            var imagesToUpdate = _k8sApi.GetImages()
+                .Where(currentClusterImage => CheckIfImageIsApplicapleForDeployment(currentClusterImage, parsedUri));
 
-            _logger.LogDebug($"Checked {ImageUriParser.ParseUri(currentClusterImage.Image)} against {parsedUri} and {nameof(isApplicaple)} returned {isApplicaple}");
+            foreach(var imageToUpdate in imagesToUpdate)
+            {
+                _k8sApi.ForceUpdateOfDeployment(imageToUpdate);
+                _logger.LogInformation($"Updated image 'deployment/{imageToUpdate.DeploymentName} {imageToUpdate.ContainerName}={parsedUri.uri}:{parsedUri.tag} --namespace={imageToUpdate.NameSpace}'");
+            }
 
-            return isApplicaple;
+            return imagesToUpdate.Select(image => new ImageEvent()
+            {
+                Image = parsedUri.uri,
+                Tag = parsedUri.tag,
+                Deployment = image.DeploymentName,
+                NameSpace = image.NameSpace,
+                TimeStamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds(),
+                Message = ResolveMessage(image)
+            });
         }
 
-        private ImageEvent SetNewImage((string uri, string tag) imageUri, ImageInCluster image)
+        private string ResolveMessage(ImageInCluster image)
         {
-            var newImage = _k8sApi.SetImage(image, $"{imageUri.uri}:{imageUri.tag}");
-            var entity = _context.EventHistory.Add(new ImageEvent()
+            if(image.ImagePullPolicy == "Always")
```

Comment:
"Always" could moved to const and same value could be used in the "error" message below. Something like "ExpectedImagePullPolicy" or something. Or not.